packages/sdk-codegen/src/sdkModels.ts

```diff
@@ -1541,10 +1541,12 @@ export class Type implements IType {
         // Track the "parent" reference for this type from the property reference
         propType.parentTypes.add(this.name)
         if (
-          propType.instanceOf('ArrayType') &&
-          propType.elementType?.instanceOf('EnumType')
+          propType.instanceOf('ArrayType') ||
+          propType.instanceOf('HashType')
         ) {
-          propType.elementType.parentTypes.add(propType.name)
+          propType.elementType?.parentTypes.add(propType.name)
+          propType.elementType?.parentTypes.add(this.name)
```

Comment:
why do we want both `DashboardElement[]` and `Dashboard`? It seems to me that `IType.{types|parentTypes}` is meant to reference other `IType` objects which I believe (though I could be wrong) are anything that would be in a `"$ref"` value from the spec. I don't think `DashboardElement[]` satisfies that requirement. In fact, in APIX you're using `typeRefs(...)` which appears to filter out anything that isn't a `"$ref"` (and thus `DashboardElement[]` is removed)?

I don't know which of these two lines puts in `DashboardElement[]` but I think it should be removed.

Caveat: not super up to speed on the actual details/usage of the `SdkModel` so I could be totally wrong here.

Reply:
`parentTypes` is primarily for APIX explorer documentation purposes to link a type to the types that use or "contain" it. For exploration, we don't bother to show an array collection type, just the underlying complex type.

Reply:
This is a good point and would only require some additional validation in sdkModels.ts. I'm unsure whether we should do that or not though, I'll have to defer to @jkaster

Reply:
I didn't answer completely. I think `DashboardElement` should have parent types that are `Dashboard` and `DashboardElement[]` because both "contain" `DashboardElement`.

Reply:
I guess my more basic question is what what is a "type" in the context here.

* `this` is an instance of `Type` that implements `IType`
* `IType.{types|parentTypes}`? is currently just `Set<string>` but it seems to me like they're more meant to be `Set<IType>` (which I know doesn't actually work under the hood due to object equality checks on a set).

~~If~~ that's correct (i.e. type/parentTypes is mean to represent a collection of IType) then what is an `IType`? A `Dashboard` is an `IType` but `DashboardElement[]` is not. The former exists as a value to a `$ref` key in the schema where as the latter never is, rather `DashboardElement` is referenced in the `items` key of a property that has a `type` value of `"array"`?

On the other hand, maybe types/parentTypes is not a loose reference to IType but rather meant to be the union of IType and IType[]? But, as you say it's primarily for APIX which promptly discards the IType[] type by calling `typeRefs(...)`?

Or maybe I'm just up too early and my brain isn't on :-) happy to hash out in person later

Reply:
Ok, I thought that might be the question. They must be `Set<string>` rather than `IType` to prevent circular references that break JSON serialization. You'll find we use the `name` of the type rather than the actual Type in several collections for this reason.

Reply:
I'm not doing a good job expressing my question/concern: it's not about the implementation of the typescript type or javascript runtime implementation of `IType.{types|parentTypes}`: my question/concern is what kinds of "things" do `IType.{types|parentTypes}` refer to. I believe they refer to `IType` things (even though there's no typescript type constraint because we're just using `Set<string>`), **not** container-datastructures of `IType` things and hence `DashboardElement[]` does not belong.

I'm thinking of an ERD - a `parentType` would be another box but would not also include the many-to-one line-arrow pointing to the box

Reply:
I guess I see `DashboardElement[]` and `Dashboard` as two fundamentally different kinds of things:

`Dashboard` - a real thing (a `User` is a real thing etc)
`DashboardElement[]` - a programming construct to describe a collection of real things

Housing those under the same property feels like saying "one of my parents is named Tom, and I have another parent whose name is 'Tom's kids'"

Reply:
I agree it gets muddy. What do you suggest instead? We definitely need to link child types of a collection to the user of said collection. For visual representation in APIX, this works. Can we create a ticket to work out this design and address it in another PR, because I've already thought about this and haven't come up with anything clearly better yet, and for the purpose of APIX type ref navigation this works.

Reply:
> We definitely need to link child types of a collection to the user of said collection

I don't totally understand what this means w/out seeing concrete code but I'll take your word for it: carry on :-)